Setting distinct_hosts to a boolean causes panic

The document states that the `distinct_hosts` attribute of the `constraints` object should be a boolean:

```hcl
# Define a job called my-service
job "consul" {
    # Job should run in the US region
    region = "global"

    # Spread tasks between us-west-1 and us-east-1
    datacenters = ["dc1"]

    # run this job globally
    type = "system"

    # Rolling updates should be sequential
    update {
        stagger = "30s"
        max_parallel = 1
    }
	
	constraint{
		distinct_hosts = true
	}
	
	group "consul-server" {
		...
	}
}
```

However, if I set it to `true`, nomad panics:
```
panic: interface conversion: interface is bool, not string

goroutine 1 [running]:
github.com/hashicorp/nomad/jobspec.parseConstraints(0xc82000cae8, 0xc8201975a0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/jobspec/parse.go:321 +0x757
github.com/hashicorp/nomad/jobspec.parseJob(0xc82000ca80, 0xc8200f9080, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/jobspec/parse.go:118 +0x502
github.com/hashicorp/nomad/jobspec.Parse(0x7f1169afc270, 0xc820026220, 0xc820026220, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/jobspec/parse.go:53 +0x404
github.com/hashicorp/nomad/jobspec.ParseFile(0xc8200f8940, 0x1a, 0x0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/jobspec/parse.go:73 +0x15c
github.com/hashicorp/nomad/command.(*RunCommand).Run(0xc8200f8900, 0xc82000a1b0, 0x1, 0x1, 0xc820106ea0)
        /opt/gopath/src/github.com/hashicorp/nomad/command/run.go:77 +0x28a
github.com/mitchellh/cli.(*CLI).Run(0xc8200b7c30, 0xc8200b7c30, 0x0, 0x0)
        /opt/gopath/src/github.com/mitchellh/cli/cli.go:112 +0x655
main.RunCustom(0xc82000a190, 0x3, 0x3, 0xc82015a390, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/main.go:47 +0x469
main.Run(0xc82000a190, 0x3, 0x3, 0xfa48e8)
        /opt/gopath/src/github.com/hashicorp/nomad/main.go:15 +0x4c
main.main()
        /opt/gopath/src/github.com/hashicorp/nomad/main.go:11 +0x60

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 8 [runnable]:
text/template/parse.lexText(0xc820083180, 0xf98208)
        /usr/local/go/src/text/template/parse/lex.go:237 +0x37e
text/template/parse.(*lexer).run(0xc820083180)
        /usr/local/go/src/text/template/parse/lex.go:206 +0x52
created by text/template/parse.lex
        /usr/local/go/src/text/template/parse/lex.go:199 +0x15d

goroutine 9 [runnable]:
github.com/hashicorp/nomad/command.(*Meta).FlagSet.func1(0xc820083280, 0xc8200f8900)
        /opt/gopath/src/github.com/hashicorp/nomad/command/meta.go:58
created by github.com/hashicorp/nomad/command.(*Meta).FlagSet
        /opt/gopath/src/github.com/hashicorp/nomad/command/meta.go:62 +0x3ce
```

--- a/jobspec/parse.go
+++ b/jobspec/parse.go
@@ -318,17 +318,9 @@ func parseConstraints(result *[]*structs.Constraint, list *ast.ObjectList) error
 		}
 
 		if value, ok := m[structs.ConstraintDistinctHosts]; ok {
-			var enabled bool
-			var err error
-			switch value.(type) {
-			case string:
-				if enabled, err = strconv.ParseBool(value.(string)); err != nil {
-					return err
-				}
-			case bool:
-				enabled = value.(bool)
-			default:
-				return fmt.Errorf("distinct_hosts should be set to true or false; got %v", value)
+			enabled, err := parseBool(value)
+			if err != nil {
+				return fmt.Errorf("distinct_hosts should be set to true or false; %v", err)
 			}
 
 			// If it is not enabled, skip the constraint.
@@ -352,6 +344,23 @@ func parseConstraints(result *[]*structs.Constraint, list *ast.ObjectList) error
 	}
 
 	return nil
+}
+
+// parseBool takes an interface value and tries to convert it to a boolean and
+// returns an error if the type can't be converted.
+func parseBool(value interface{}) (bool, error) {
+	var enabled bool
+	var err error
+	switch value.(type) {
+	case string:
+		enabled, err = strconv.ParseBool(value.(string))
+	case bool:
+		enabled = value.(bool)
+	default:
+		err = fmt.Errorf("%v couldn't be converted to boolean value", value)
+	}
+
+	return enabled, err
 }
 
 func parseTasks(jobName string, taskGroupName string, result *[]*structs.Task, list *ast.ObjectList) error {

--- a/jobspec/parse.go
+++ b/jobspec/parse.go
@@ -318,9 +318,17 @@ func parseConstraints(result *[]*structs.Constraint, list *ast.ObjectList) error
 		}
 
 		if value, ok := m[structs.ConstraintDistinctHosts]; ok {
-			enabled, err := strconv.ParseBool(value.(string))
-			if err != nil {
-				return err
+			var enabled bool
+			var err error
+			switch value.(type) {
+			case string:
+				if enabled, err = strconv.ParseBool(value.(string)); err != nil {
+					return err
+				}
+			case bool:
+				enabled = value.(bool)
+			default:
+				return fmt.Errorf("distinct_hosts should be set to true or false; got %v", value)
 			}
 
 			// If it is not enabled, skip the constraint.

